### PR TITLE
Extend bitcoin networks to all address types

### PR DESCRIPTION
**Pull request by Kailai-Wang**
### Context

As topic.

Theoretically the client only needs to rename `Bitcoin` to `BitcoinP2tr` and that's all

### CODE REVIEW

**Review by jonalvarezz — APPROVED** [review]
looking good to me

File: primitives/core/src/network.rs [hunk 1]
```diff
@@ -70,9 +70,17 @@ pub enum Web3Network {
 	#[codec(index = 8)]
 	Bsc,
 
-	// btc
+	// btc, see https://github.com/rust-bitcoin/rust-bitcoin/blob/9ea3e29d61569479b7b4618c8ae1992612f3d01a/bitcoin/src/address/mod.rs#L64-L75
 	#[codec(index = 9)]
-	Bitcoin,
+	BitcoinP2tr,
```
Comment (kziemianek) [inline primitives/core/src/network.rs:75]: Have you considered using nested enum? I think there is only one bitcoin network 🙃

Comment (Kailai-Wang) [reply primitives/core/src/network.rs:75]: there's only one mainnet but these are used to distinguish between different address types, so theoretically one pubkey can be used for any one of the address types. From this sense it's similar to EVM where one universal pubkey for all networks. Wdyt

Comment (Kailai-Wang) [reply primitives/core/src/network.rs:75]: I'll add a comment later

**Pull request merged**